### ISSUE TITLE
Existing profile pictures not shown after update to 1.3.0.

### ISSUE THREAD

**Issue by raimund-schluessler** [owncloud/contacts#421]
### Steps to reproduce
1. Have a contact with already existing profile picture (set with old contacts app or imported from google one day)
2. Upgrade contacts app form 1.2.0.0 to 1.3.0.0
3. Have a look at the contacts profile picture
### Expected behaviour

The profile picture should be shown.
### Actual behaviour

The image is not shown, there is a problem loading the image although there is an encoded image present in the code.
### Server configuration

**Operating system**: Debian Jessie

**Web server:** Apache 2.4

**Database:** MySQL

**PHP version:** 5.6

**ownCloud version:** 9.0.2

**Contacts version:** 1.3.0.0

**Updated from an older ownCloud or fresh install:** Updated from contacts 1.2.0.0

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
- Standard apps
- Calendar
- Contacts
- Tasks
- Mail
```

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no
### Client configuration

**Browser:** Firefox ESR 45.1.0

**Operating system:** Windows 10
#### Web server error log

```
I don't have access at the moment. Will hand in later if necessary.
```
#### ownCloud log (data/owncloud.log)

```
I don't have access at the moment. Will hand in later if necessary.
```
#### Browser log

```
No errors or warnings in the log.
```

**Comment by raimund-schluessler** [comment]
When I click on the respective contact I see that there is a problem with CSP preventing the image form loading:

```
Content Security Policy: The page's settings blocked the loading of a resource at unsafe:data:JPEG;base64,[trimmed for security and clarity]
```

However, the embedded preview picture in the list is not loaded either.

**Comment by raimund-schluessler** [comment]
The problem of the profile picture in the list is due to: #315.
When manually changing

```
src="data:image/png;base64,data:JPEG;base64,
```

to

```
src="data:image/png;base64,[imagedata]
```

the image is shown correctly.

So this image here is only about the CSP issue. Any ideas? Is my server configured wrong?

**Comment by raimund-schluessler** [comment]
When I create a new contact and upload a picture, it is correctly shown in the contact details, but still not shown in the preview list.

**Comment by DeepDiver1975** [comment]
Can I ask you to post a sample vcard for debugging? THX

**Comment by DarkSteve** [comment]
I'm experiencing the same thing. My server configuration is identical except I'm on FreeBSD 10.3.

Who'd have thought the Contact app could be so problematic!

**Comment by DeepDiver1975** [comment]
> Who'd have thought the Contact app could be so problematic!

What are you trying to tell me?

**Comment by DeepDiver1975** [comment]
@Henni this seems to be causing this issue - https://github.com/owncloud/contacts/compare/v1.2.0.0...v1.3.0.0#diff-25fe07d2d32ea4ff3c47532c18614c20R97

**Comment by DarkSteve** [comment]
> What are you trying to tell me?

Nothing bad! I didn't mean to offend!

It's just that since upgrading to ownCloud 9, the new Contacts app has had a few very hard to squash bugs. For me, importing vCards and ongoing profile picture issues have been the most prominent.

It's unfortunate, but not unexpected. I mean, the app had to be pretty much rewritten from scratch. I think you guys are doing great work!

**Comment by raimund-schluessler** [comment]
@DeepDiver1975 You can find a vcf file here: https://gist.github.com/raimund-schluessler/b18d453873afc7d3cf2d0b5e55b65d41
I replaced sensitive information, but it seems as the vcf is still intact (reimport into contacts app worked).

**Comment by DeepDiver1975** [comment]
THX @raimund-schluessler

**Comment by j-ed** [comment]
@DeepDiver1975 your fix https://github.com/owncloud/contacts/commit/e645ade626852f025b3223e9ea7d7bd783047566 modifies file `js/models/contact_model.js` which is currently not part of the 1.3.0.0 app archive. Is that correct or has an incomplete archive been published? I would like to apply the patch manually if possible.

**Comment by raimund-schluessler** [comment]
@j-ed You have to edit line 1037 in `js/public/script.js`. This file is compiled from the source files in this repository here. However, I must admit, that I don't really understand why the compiled version is not part of the repo.

@DeepDiver1975 Is there a reason why the compiled version is not tracked here? I know it might be a bit redundant to track it, but it would enable people to test the app directly without installing any dev tools.

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 Is there a reason why the compiled version is not tracked here? I know it might be a bit redundant to track it, but it would enable people to test the app directly without installing any dev tools.

Merge conflicts and the constant failure of devs to regenerate the compiled file on checkin :wink:

**Comment by j-ed** [comment]
@raimund-schluessler :+1: Thanks, I've applied the patch https://github.com/owncloud/contacts/commit/e645ade626852f025b3223e9ea7d7bd783047566 and the contact pictures are shown again.

**Comment by ManOki** [comment]
i uploaded an photo again and all changed in the vcard photo section is:

OLD: `PHOTO;ENCODING=B;TYPE=JPEG:/9j/4AAQSkZJRg`
NEW: `PHOTO;TYPE=image/jpeg:/9j/4AAQSkZJRg`

no need to update, just reupload all images or update database.

Note: i had very old contact images from owncloud 6, same as mentioned in #315 

update: the type of the new vcf export does not seem valid (see [wiki](https://en.wikipedia.org/wiki/VCard#Properties)) ... vcard version is still 3.0 and there is 'officially' no such thing as `TYPE=image/jpeg`, just `TYPE=jpeg`

**Comment by raimund-schluessler** [comment]
Uploading a new picture only solves this issue here (#421) and it is very cumbersome to do it for all contacts.

The fix provided in #422 solves both this issue here and #315 and is only a small fix. So I definitly go with that. Thanks a lot @DeepDiver1975
